options/options.go

```diff
@@ -2215,3 +2250,32 @@ func newSrcInfo(path []int32, children sourceinfo.OptionChildrenSourceInfo) sour
 		Children: children,
 	}
 }
+
+type lenientHandler interpreter
```

Comment:
Maybe `*reporter.Handler` could have a lenient method to enable and disable modes, removing the need to keep state in the interpreter?

Reply:
The handler is thread-safe and even supports a handler "hierarchy", so trying to fit this extra state into it would make its implementation more complicated. Also, there is no provided handler for cases where we can be lenient, so it would be adding exported API to the handler types for no real value (other than possibly as a slight convenience to this package), which seems like the wrong tradeoff.